### ISSUE TITLE
[Bug]: dump macro_meta cause  ob_admin crash

### ISSUE THREAD

**Issue by beifangxiuwhx** [oceanbase/oceanbase#1095]
### Please check the issue list for the same bug

- [X] Please checked issue list and confirm this bug is encountered for the first time.
- [X] Please try full text in English and attach precise description.

### Environment

```markdown
- OB Version&commit_id or other module version&commit_id (e.g. observer V3.1.2 8b23a93):
- Hardware parameter:OS type & Compliler & CPU etc. 
- Others

observer V3.1.4 in all operation system
```

### Fast Reproduce Steps(Required)

use ob_admin to dump macro meta information will cause ob_admin crash

### Actual Behavior

ob_admin will crash

### Expected Behavior

it should dump macro_meta info

### other information

access nil pointer cause process crash,  just in  `print_macro_meta`  function **meta**

```. 

void ObAdminDumpsstExecutor::print_macro_meta()
{
  // int ret = OB_SUCCESS;
  const ObMacroBlockMeta *meta = NULL;
  MacroBlockId macro_id(dump_macro_context_.macro_id_);
  PrintHelper::print_dump_title("Macro Meta");
  PrintHelper::print_dump_line("macro_block_id", dump_macro_context_.macro_id_);
  PrintHelper::print_dump_line("attr", meta->attr_);
  PrintHelper::print_dump_line("data_version", meta->data_version_);
  PrintHelper::print_dump_line("column_number", meta->column_number_);
  PrintHelper::print_dump_line("rowkey_column_number", meta->rowkey_column_number_);
  PrintHelper::print_dump_line("column_index_scale", meta->column_index_scale_);
  PrintHelper::print_dump_line("row_store_type", meta->row_store_type_);
  ........
```

**Comment by JLY2015** [comment]
Do you need to use the print_macro_meta function? 
This function is deprecated, but the code has not been deleted

**Comment by beifangxiuwhx** [comment]
Yes， dumping file/meta is necessary for ops engineer .

And documentations in ob community have no mention of expiration  about "dumpsst -d mm".

I think it‘s inspiring for users if community can complete ob_admin tools。

**Comment by beifangxiuwhx** [comment]
> Do you need to use the print_macro_meta function? This function is deprecated, but the code has not been deleted

Yes， dumping file/meta is necessary for ops engineer .

And documentations in ob community have no mention of expiration about "dumpsst -d mm".

I think it‘s inspiring for users if community can complete ob_admin tools

**Comment by dimstars** [comment]
fixed in 4.0
